apparelo/apparelo/doctype/dc/dc.py

```diff
@@ -57,6 +58,10 @@ def on_submit(self):
 		msgprint(_("{0} created").format(comma_and(
 			"""<a href="#Form/Stock Entry/{0}">{1}</a>""".format(stock_entry.name, stock_entry.name))))
 
+	def on_cancel(self):
+		self.db_set("docstatus",2)
+		msgprint(_("{0} cancelled").format(comma_and("""<a href="#Form/DC/{0}">{1}</a>""".format(self.name, self.name))))
+		
```

Comment:
don't we also need to cancel the relevant PO and SE? I don't see that happening here... Am I missing something here?

Reply:
Yes, I didn't write any code to cancel corresponding PO and SE. It works fine here.Because DC is linked to  PO and SE.So while cancelling DC it automatically fetch the linked PO and  SE with cancel all option as in the above screenshots.After all linked documents are cancelled I change the docstatus of DC and show a pop up message.